wordpress-mobile/WordPress-iOS
# In-App Notifications (Notices) #1

This PR implements the first part of in-app notifications, to be used in the media library and plugins sections of the app. In this PR, we're adding:

* A WordPressFlux store for Notices, and its accompanying Notice type and actions.
* A temporary Notice Presenter, which uses UIAlertController to display notices. This will be replaced later with custom UI.
* MediaCoordinator posts a notification when a set of uploads complete.
* The feature is feature-flagged to local develop only.

### Known Issues / Not Implemented

* Because the Notice Presenter is temporary, there may be a couple of issues using UIAlertControllers to present notices – for example, they won't be presented if another view controller is already being presented modally.
* The media library currently only presents a notification on success. The failure case will be handled later.
* The action button on the alert (Write Post) currently doesn't do anything.

To test:

* Open the Media Library and upload a couple of images.
* Ensure that an alert is displayed when all the uploads complete. 
* Ensure that the alert is still displayed if you leave the media library during uploads.
* Also check uploading more images after the initial set complete.
* Modify `mediaProgressCoordinatorDidFinishUpload` in `MediaCoordinator` to dispatch multiple notices one after the other. Check that alerts are queued correctly (with the next appearing after you dismiss one).
